src/TestUtils/src/DeviceTests/AssertionExtensions.Windows.cs

```diff
@@ -59,34 +62,49 @@ public static async Task<T> AttachAndRun<T>(this FrameworkElement view, Func<Tas
 			if (view.Parent is Border wrapper)
 				view = wrapper;
 
-			// TODO
-
-			//var layout = new FrameLayout(view.Context)
-			//{
-			//	LayoutParameters = new FrameLayout.LayoutParams(ViewGroup.LayoutParams.MatchParent, ViewGroup.LayoutParams.MatchParent)
-			//};
-			//view.LayoutParameters = new FrameLayout.LayoutParams(view.Width, view.Height)
-			//{
-			//	Gravity = GravityFlags.Center
-			//};
-
-			//var act = view.Context.GetActivity();
-			//var rootView = act.FindViewById<FrameLayout>(Android.Resource.Id.Content);
-
-			//layout.AddView(view);
-			//rootView.AddView(layout);
-
-			//await Task.Delay(100);
-
-			try
+			if (view.Parent == null)
 			{
-				var result = await action();
-				return result;
+				Grid grid;
+				var window = new Window
+				{
+					Content = new Grid
+					{
+						HorizontalAlignment = HorizontalAlignment.Center,
+						VerticalAlignment = VerticalAlignment.Center,
+						Children =
+					{
+						(grid = new Grid
+						{
+							Width = view.Width,
+							Height = view.Height,
+							Children =
+							{
+								view
+							}
+						})
+					}
+					}
+				};
+				window.Activate();
+
+				try
+				{
+					return await Run(action);
+				}
```

Comment:
@PureWeen I fixed the AttachAndRun for windows, so now it pops up a new window with the control. Much nicer than hacking the current UI.